java/maven/src/org/netbeans/modules/maven/classpath/ClassPathProviderImpl.java

```diff
@@ -721,18 +722,14 @@ public AnnotationPathSelector(NbMavenProjectImpl proj, ClassPath anno, Supplier<
             this.annotationCP = anno;
             this.compileClassPath = compile;
             
-            anno.addPropertyChangeListener(WeakListeners.propertyChange(
-                    e -> {
-                        active = null;
-                        support.firePropertyChange(PROP_ACTIVE_CLASS_PATH, null, null);
-                    }, ClassPath.PROP_ROOTS, anno
-            ));
-//            proj.getProjectWatcher().addPropertyChangeListener((e) -> {
-//                if (NbMavenProject.PROP_PROJECT.equals(e.getPropertyName())) {
-//                    active = null;
-//                    support.firePropertyChange(PROP_ACTIVE_CLASS_PATH, null, null);
-//                }
-//            });
+            anno.addPropertyChangeListener(WeakListeners.propertyChange(this, anno));
+            proj.getProjectWatcher().addPropertyChangeListener(WeakListeners.propertyChange(this, anno));
```

Comment:
Should be the 2nd parameter the event source, `proj.getProjectWatcher()` in this case ?

Reply:
it was 'anno' before the change. Since this is a last minute fix, it's probably safer to not change this too, right?

Reply:
let me check this in the ide in case i didn't count the brackets properly

Reply:
@mbien from the docs, the property name should be passed to both `WeakListeners::propertyChange` and `addPropertyChangeListener`. As not used for the latter, I assume this might have been firing for all properties but only trying to remove a listener on `ClassPath.PROP_ROOTS`?!

@sdedic is right, that second weak listener should be passed `proj::getProjectWatcher`, or `null`.  I think the underlying code can handle events from alternative sources though, so shouldn't cause a problem?

I (or one of you) can add a second commit here to tidy this up, but need a call ASAP if we're going to get in rc3.

cc @timboudreau

Reply:
@sdedic added a second commit that fixes the source passed to WeakListeners.  Other than that are you happy with @timboudreau changes going in to 16-rc3?

Reply:
Is there a reason to react on all propchanges on project, not only PROP_PROJECT (= reload of project file) ?

Reply:
@sdedic no idea. Justification for that is covered by Tim in description for #4868

Reply:
@sdedic have merged to proceed with rc3.  If you have concerns about this reacting to too many changes, we can review before release vote.